### PR TITLE
[skip gpuci] Add GitHub Actions workflows

### PR DESCRIPTION
**Pull request by Ethyling**
This PR adds the first version of the GitHub Actions workflows for `rmm`

### CODE REVIEW

File: ci/check_style.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+set -euo pipefail
+
+rapids-logger "Create checks conda environment"
+# TODO: Move this
+. /opt/conda/etc/profile.d/conda.sh
```
Comment (ajschmidt8) [inline ci/check_style.sh:8]: do we still need to move this? if not, can we remove this comment?

File: ci/test_cpp.sh [hunk 2]
```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -euo pipefail
+
+# TODO: Move this
+. /opt/conda/etc/profile.d/conda.sh
+conda activate base
```
Comment (ajschmidt8) [inline ci/test_cpp.sh:7]: same question.

do we still need to move this? if not, can we remove this comment?

looks like there's a few other instances of this in the PR as well.

Comment (ajschmidt8) [reply ci/test_cpp.sh:7]: should we move this to a `rapids-activate-base-env` script in `gha-tools`?

File: ci/test_cpp.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+# Check environment
```
Comment (ajschmidt8) [inline ci/test_cpp.sh:9]:
```suggestion
```

What do you think about removing all extraneous comments like this?

File: ci/test_python.sh [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+rapids-logger "Create test conda environment"
+# TODO: Move this
+. /opt/conda/etc/profile.d/conda.sh
+
+rapids-dependency-file-generator \
+  --generate conda \
+  --file_key test \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" > env.yaml
+
+mamba env create -f env.yaml -n test
```
Comment (ajschmidt8) [inline ci/test_python.sh:14]:
```suggestion
rapids-mamba-retry env create -f env.yaml -n test
```

File: ci/check_style.sh [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+rapids-dependency-file-generator \
+  --generate conda \
+  --file_key checks \
+  --matrix "cuda=${RAPIDS_CUDA_VERSION%.*};arch=$(arch);py=${RAPIDS_PY_VERSION}" > env.yaml
+
+mamba env create -f env.yaml -n checks
```
Comment (ajschmidt8) [inline ci/check_style.sh:15]:
```suggestion
rapids-mamba-retry env create -f env.yaml -n checks
```

This step seems to fail sometimes without `rapids-mamba-retry`. I added a retry for `ConnectionError`s [here](https://github.com/rapidsai/gha-tools/commit/eb4a92c087e3f5789628b6636048b75b8dce8311)

- https://github.com/rapidsai/rmm/actions/runs/3136315901/jobs/5093520359

**Comment by Ethyling** [comment]
Closed in favor of https://github.com/rapidsai/rmm/pull/1104

**Pull request closed without merging**